### PR TITLE
Use Network Provisioning for both chip-tool and ChipTool iOS

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Problem
As we start to migrate from current Rendezvous Network Provisioning API to Network Provisioning cluster. Several previous Rendezvous code paths are broken. This might impact existing the iOS chip-tool. We need to make sure chip-tool, and the iOS app are not broken because of this change.

 #### Summary of Changes
Use Network Provisioning for both chip-tool and ChipTool iOS.

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

File: examples/chip-tool/commands/pairing/PairingCommand.h [hunk 1]
```diff
@@ -69,19 +93,51 @@ class PairingCommand : public Command, public chip::Controller::DevicePairingDel
...
+    chip::Callback::Callback<NetworkCommissioningClusterEnableNetworkResponseCallback> * mOnEnableNetworkCallback;
+    chip::Callback::Callback<DefaultFailureCallback> * mOnFailureCallback;
     ChipDeviceCommissioner mCommissioner;
+    ChipDevice * mDevice;
+    chip::Controller::NetworkCommissioningCluster mCluster;
+
+    // In theory the endpoint should be 0, or alternatively it should be found using the descriptor
+    // cluster.
+    // But at the moment, endpoint starts from 1.
+    chip::EndpointId mEndpointId = 1;
```
Comment (vivien-apple) [inline examples/chip-tool/commands/pairing/PairingCommand.h:142]: Looking at #6000 I guess `1` should be replaced by `0` if #6000 lands before this one.

**Review by woody-apple — CHANGES_REQUESTED** [review]
Per @vivien-apple 's feedback

**Comment by yufengwangca** [comment]
> Per @vivien-apple 's feedback

I will wait for #6000 land and make the change

**Comment by yufengwangca** [comment]
> > Per @vivien-apple 's feedback
> 
> I will wait for #6000 land and make the change

@vivien-apple @woody-apple updated, pls check

**Review by woody-apple — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**